org.opentosca.bus.management.service.impl/src/org/opentosca/bus/management/service/impl/ManagementBusServiceImpl.java

```diff
@@ -135,13 +135,25 @@ public void invokeIA(final Exchange exchange) {
         if (!serviceTemplateInstanceID.equals(Long.MIN_VALUE)) {
 
             if (Boolean.valueOf(Settings.OPENTOSCA_BUS_MANAGEMENT_MOCK)) {
-                
+
                 long waitTime = System.currentTimeMillis() + 1000;
-                while(System.currentTimeMillis() < waitTime) {                
+                while (System.currentTimeMillis() < waitTime) {
                 }
-                
-                respondViaMocking(exchange, csarID, serviceTemplateID, nodeTemplateID, neededInterface,
-                                  neededOperation);
+
+                // if some param has fault as a value we generate a fault response
+                final Object params = message.getBody();
+                if (params != null && params instanceof HashMap && ((HashMap) params).values().contains("fault")) {                    

```

Comment:
this can be pushed down into `respondViaMocking`. AFAICT the only difference between `respondViaFault` and `respondViaMocking` is the addition of a `"Fault"` key in the response map.